### PR TITLE
RR-493 - Tidy up entity mappers

### PR DESCRIPTION
**Pull request by mattwills-moj-digital**
This PR reduces duplication in the entity mappers and makes some improvements that I've explained in comments below.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/induction/FutureWorkInterestsEntityMapper.kt [hunk 1]
```diff
@@ -62,33 +61,38 @@ abstract class FutureWorkInterestsEntityMapper {
...
   @Mapping(target = "updatedAtPrison", source = "prisonId")
   @Mapping(target = "interests", ignore = true)
   abstract fun fromUpdateDtoToEntity(dto: UpdateFutureWorkInterestsDto?): FutureWorkInterestsEntity
 
+  @Named("addNewInterestsDuringUpdate")
   @AfterMapping
-  fun addInterests(dto: UpdateFutureWorkInterestsDto, @MappingTarget entity: FutureWorkInterestsEntity) {
-    val existingTypes = entity.interests?.map { it.key() }
-    dto.interests.forEach {
-      // ensure we only add new interests
```
Comment (mattwills-moj-digital) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/induction/FutureWorkInterestsEntityMapper.kt:76]: A colleague of mine (😉 ) pointed out that you can control which client method calls an `@AfterMapping` method. Upon reading the MapStruct docs (🙄 ), I've discovered that using `@BeanMapping(qualifiedByName = ["addNewInterestsDuringUpdate"])` on the originating method, means that `addNewInterestsDuringUpdate` is only called from `fromUpdateDtoToEntity()` and not from `updateEntityFromDto()` as well. This means we no longer need to check if a child entity already exists in the collection before adding it.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/induction/FutureWorkInterestsEntityMapper.kt [hunk 2]
```diff
@@ -29,21 +31,18 @@ abstract class FutureWorkInterestsEntityMapper {
...
   @AfterMapping
-  fun addInterests(dto: CreateFutureWorkInterestsDto, @MappingTarget entity: FutureWorkInterestsEntity) {
-    dto.interests.forEach {
-      entity.addChild(
-        workInterestEntityMapper.fromDomainToEntity(it),
-        entity.interests(),
-      )
-    }
+  fun addNewInterestsDuringCreate(dto: CreateFutureWorkInterestsDto, @MappingTarget entity: FutureWorkInterestsEntity) {
+    addNewInterests(dto.interests, entity)
```
Comment (mattwills-moj-digital) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/induction/FutureWorkInterestsEntityMapper.kt:45]: A simple shared method to add new interests, either when creating an Induction (i.e. via `CreateFutureWorkInterestsDto`) or when updating it (via `UpdateFutureWorkInterestsDto`).

File: src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/induction/FutureWorkInterestsEntityMapper.kt [hunk 3]
```diff
@@ -62,33 +61,38 @@ abstract class FutureWorkInterestsEntityMapper {
     dto: UpdateFutureWorkInterestsDto?,
   )
 
+  fun updateInterests(entity: FutureWorkInterestsEntity, dto: UpdateFutureWorkInterestsDto): List<WorkInterestEntity> {
```
Comment (mattwills-moj-digital) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/educationandworkplanapi/app/database/jpa/mapper/induction/FutureWorkInterestsEntityMapper.kt:64]: I've just moved this from below to place it next to the method that calls it and also to make it consistent with the other mappers.

**Review by nathanrussell-moj-digital — APPROVED** [review]
What is not to love? 😁 
LGTM 👍

**Pull request merged**